lib/registry-secrets.ts

```diff
@@ -89,3 +95,78 @@ export class RegistrySecretValidator {
 		return this.validateRegistrySecrets(JSON.parse(json));
 	}
 }
+
+/**
+ * Search `registrySecrets` for several possible default domain names that
+ * were adopted by Docker Hub over time ('index.docker.io', 'idx.docker.io',
+ * 'docker.io', 'cloud.docker.com', 'docker.com') and clone the entry with
+ * the "canonical name" for use with the Dockerode `registryconfig` option
+ * ('https://index.docker.io/v1/'). Docker's own golang source
+ * code has some hardcoded alternatives:
+ * https://github.com/docker/distribution/blob/release/2.7/reference/normalize.go#L14
+ *
+ * By example:
+ *   Input: { 'docker.io': { 'username': 'bob',   'password': 'dog' }}
+ *   Output:
+ *   {
+ *     'docker.io': { 'username': 'bob',   'password': 'dog' },
+ *     'https://index.docker.io/v1/': { 'username': 'bob',   'password': 'dog' },
+ *   }
+ */
+export function addCanonicalDockerHubEntry(registryconfig: RegistrySecrets) {
```

Comment:
This new function will be imported in `resin-builder` in a PR I'm putting together. It will improve the user experience for the Dockerfile `FROM` instruction as well (besides the docker-compose `image:` instruction), in relation to the key (as in key-value pairs) that CLI users are required to use to identify the Docker Hub registry. Currently they are required to use exactly `'https://index.docker.io/v1/'`, but after this change they'll be allowed to use any URL container any of the following 5 domains below, plus the empty string for good measure.